src/Model.cc

```diff
@@ -792,11 +796,34 @@ const NestedInclude *Model::InterfaceModelNestedIncludeByIndex(
 }
 
 /////////////////////////////////////////////////
-sdf::ElementPtr Model::ToElement() const
+std::string Model::Uri() const
+{
+  return this->dataPtr->uri;
+}
+
+/////////////////////////////////////////////////
+void Model::SetUri(const std::string &_uri)
+{
+  this->dataPtr->uri = _uri;
+}
+
+/////////////////////////////////////////////////
+sdf::ElementPtr Model::ToElement(bool _useUri) const
 {
+  if (_useUri && !this->dataPtr->uri.empty())
+  {
+    sdf::ElementPtr worldElem(new sdf::Element);
+    sdf::initFile("world.sdf", worldElem);
```

Comment:
Does this work okay if the model is a nested model?

Reply:
Added a test for this in fbadb2d.